azurerm/data_source_container_orchestrator_versions.go

```diff
@@ -0,0 +1,87 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/go-version"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceArmContainerOrchestratorVersions() *schema.Resource {
+	return &schema.Resource{
+		Read: dataArmSourceContainerOrchestratorVersionsRead,
+
+		Schema: map[string]*schema.Schema{
+			"location": locationSchema(),
+			"orchestrator_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "Kubernetes",
+			},
+			"version_prefix": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"versions": {
+				Type:     schema.TypeList,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Computed: true,
+			},
+			"latest_version": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataArmSourceContainerOrchestratorVersionsRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).containerServicesClient
+	ctx := meta.(*ArmClient).StopContext
+
+	location := d.Get("location").(string)
+	ot := d.Get("orchestrator_type").(string)
+
+	listResp, err := client.ListOrchestrators(ctx, location, "managedClusters")
+	if err != nil {
+		if utils.ResponseWasNotFound(listResp.Response) {
+			return fmt.Errorf("Error: Container Orchestrators not found for location %q", location)
```

Comment:
what's the use-case for this, when AKS isn't supported in that region?

Reply:
I would say yes, but I'm not 100% sure this is the right error to check for that case. Any idea how to confirm it?

The API docs only mention positive responses. Does that mean there can be no errors returned?
https://docs.microsoft.com/en-us/rest/api/container-service/container%20service%20client/listorchestrators

Reply:
I'm 99% sure that'll come through as a 404; it's generally something along the lines of `the API version '2019-01-01' was not found for the Resource Provider 'Microsoft.ContainerService'`